frame/nis/src/lib.rs

```diff
@@ -0,0 +1,925 @@
+				b"proportion" => Some(item.proportion.encode()),
+				b"expiry" => Some(item.expiry.encode()),
+				_ => None,
+			}
+		}
+	}
+
+	impl<T: Config> NonfungibleTransfer<T::AccountId> for Pallet<T> {
+		fn transfer(index: &ReceiptIndex, destination: &T::AccountId) -> DispatchResult {
+			let mut item = Receipts::<T>::get(index).ok_or(TokenError::UnknownAsset)?;
+			let from = item.who;
+			item.who = destination.clone();
+			Receipts::<T>::insert(&index, &item);
+			Pallet::<T>::deposit_event(Event::<T>::Transferred {
+				from,
+				to: item.who,
+				index: *index,
+			});
+			Ok(())
+		}
+	}
+
+	impl<T: Config> Pallet<T> {
+		/// The account ID of the reserves.
+		///
+		/// This actually does computation. If you need to keep using it, then make sure you cache
+		/// the value and only call this once.
+		pub fn account_id() -> T::AccountId {
+			T::PalletId::get().into_account_truncating()
+		}
+
+		/// Returns information on the issuance within the system.
+		pub fn issuance() -> IssuanceInfo<BalanceOf<T>> {
+			Self::issuance_with(&Self::account_id(), &Summary::<T>::get())
+		}
+
+		/// Returns information on the issuance within the system
+		///
+		/// This function is equivalent to `issuance`, except that it accepts arguments rather than
+		/// queries state. If the arguments are already known, then this may be slightly more
+		/// performant.
+		///
+		/// - `our_account`: The value returned by `Self::account_id()`.
+		/// - `summary`: The value returned by `Summary::<T>::get()`.
+		pub fn issuance_with(
+			our_account: &T::AccountId,
+			summary: &SummaryRecordOf<T>,
+		) -> IssuanceInfo<BalanceOf<T>> {
+			let total_issuance =
+				T::Currency::total_issuance().saturating_sub(T::IgnoredIssuance::get());
+			let holdings = T::Currency::free_balance(our_account);
+			let other = total_issuance.saturating_sub(holdings);
+			let effective =
+				summary.proportion_owed.left_from_one().saturating_reciprocal_mul(other);
+			let required = summary.proportion_owed * effective;
+			IssuanceInfo { holdings, other, effective, required }
+		}
+
+		/// Process some bids into receipts up to a `target` total of all receipts.
+		///
+		/// Touch at most `max_queues`.
+		///
+		/// Return the weight used.
+		pub(crate) fn process_queues(
+			target: Perquintill,
+			max_queues: u32,
+			max_bids: u32,
+			weight: &mut WeightCounter,
+		) {
+			let mut summary: SummaryRecordOf<T> = Summary::<T>::get();
+			if summary.proportion_owed >= target {
+				return
+			}
+
+			let now = frame_system::Pallet::<T>::block_number();
+			let our_account = Self::account_id();
+			let issuance: IssuanceInfoOf<T> = Self::issuance_with(&our_account, &summary);
+			let mut remaining = target.saturating_sub(summary.proportion_owed) * issuance.effective;
+
+			let mut queues_hit = 0;
+			let mut bids_hit = 0;
+			let mut totals = QueueTotals::<T>::get();
+			let queue_count = T::QueueCount::get();
+			totals.bounded_resize(queue_count as usize, (0, Zero::zero()));
```

Comment:
Currently, you can enlarge it (add more time) without a migration. With this, you could not.

Reply:
(This code doesn't work as `bounded_vec` is only available in `std`)

Reply:
I would take it a step further and see if there's a way to override the `get` method so that we always `Get` the underlying data and `bounded_resize`.